daemon/rm.go

```diff
@@ -36,19 +36,30 @@ func (daemon *Daemon) CleanPod(podId string) (int, string, error) {
 		}
 	}
 
-	os.RemoveAll(path.Join(utils.HYPER_ROOT, "services", podId))
-	pod.cleanupEtcHosts()
-	os.RemoveAll(path.Join(utils.HYPER_ROOT, "hosts", podId))
-
-	daemon.db.DeletePod(podId)
-	daemon.RemovePod(podId)
-	if pod.status.Type != "kubernetes" {
-		daemon.RemovePodContainer(pod)
+	pod.Lock()
+	defer pod.Unlock()
+	if pod.vm != nil {
+		pod.status.Status = types.S_POD_NONE
+		return code, cause, err
 	}
-	daemon.DeleteVolumeId(podId)
-	code = types.E_OK
 
-	return code, cause, nil
+	return daemon.RemovePodResource(pod)
+}
+
+func (daemon *Daemon) RemovePodResource(p *Pod) (int, string, error) {
+	os.RemoveAll(path.Join(utils.HYPER_ROOT, "services", p.id))
+	os.RemoveAll(path.Join(utils.HYPER_ROOT, "hosts", p.id))
```

Comment:
https://github.com/hyperhq/hyperd/pull/301/files#diff-cfb48529ee9117af9bd2e69c80e66291R25

every time pod stopped, this method will be called. No matter whether remove the pod.